core/chain-configs/src/genesis_validate.rs

```diff
@@ -1,18 +1,22 @@
 use std::collections::{HashMap, HashSet};
 
 use crate::genesis_config::{Genesis, GenesisConfig};
+use near_config_utils::{ValidationError, ValidationErrors};
 use near_crypto::key_conversion::is_valid_staking_key;
 use near_primitives::state_record::StateRecord;
 use near_primitives::types::AccountId;
 use num_rational::Rational32;
 
-/// Validate genesis config and records. Panics if genesis is ill-formed.
-pub fn validate_genesis(genesis: &Genesis) {
-    let mut genesis_validator = GenesisValidator::new(&genesis.config);
+/// Validate genesis config and records. Returns ValidationError if semantic checks of genesis failed.
+pub fn validate_genesis(genesis: &Genesis) -> Result<(), ValidationError> {
+    let mut validation_errors = ValidationErrors::new();
+    let mut genesis_validator = GenesisValidator::new(&genesis.config, &mut validation_errors);
+    println!("\nValidating Genesis config and records, extracted from genesis.json. This could take a few minutes...");
```

Comment:
Records don't have to be in genesis.json, they can be in a separate records file, which is usually records.json.
```suggestion
    tracing::info!(target: "config", "Validating Genesis config and records. This could take a few minutes...");
```